File: camionesapp/camionesapp/lib/models/camion.dart
class Camion {
  final int? id;
  final String placa;
  final String tipoCamion;
  final int? usuarioId;

  Camion({
    this.id,
    required this.placa,
    required this.tipoCamion,
    this.usuarioId,
  });

  factory Camion.fromJson(Map<String, dynamic> json) {
    return Camion(
      id: json['id'],
      placa: json['placa'],
      tipoCamion: json['tipoCamion'],
      usuarioId: json['usuarioId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'placa': placa,
      'tipoCamion': tipoCamion,
      'usuarioId': usuarioId,
    };
  }
}

File: backend/src/main/java/com/camiones/model/Camion.java
package com.camiones.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "camiones")
public class Camion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String placa;

    @NotBlank
    @Column(nullable = false)
    private String tipoCamion;

    @ManyToOne
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    public Camion() {
    }

    public Camion(String placa, String tipoCamion, Usuario usuario) {
        this.placa = placa;
        this.tipoCamion = tipoCamion;
        this.usuario = usuario;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public String getTipoCamion() {
        return tipoCamion;
    }

    public void setTipoCamion(String tipoCamion) {
        this.tipoCamion = tipoCamion;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }
}
